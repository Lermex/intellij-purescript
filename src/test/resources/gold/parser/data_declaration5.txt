Purescript File
  PSASTWrapperElement(Program)
    PSASTWrapperElement(Module)
      PsiElement(module)('module')
      PsiElement(whitespace)(' ')
      PSASTWrapperElement(ModuleName)
        PSASTWrapperElement(Qualified)
          PsiElement(proper name)('A')
        PsiElement(whitespace)(' ')
      PsiElement(where)('where')
      PsiElement(whitespace)('\n\n')
      PSASTWrapperElement(DataDeclaration)
        PsiElement(data)('data')
        PsiElement(whitespace)(' ')
        PSASTWrapperElement(TypeConstructor)
          PSProperNameImpl(ProperName)
            PsiElement(proper name)('User5')
            PsiElement(whitespace)(' ')
        PSASTWrapperElement(TypeArgs)
          PSASTWrapperElement(GenericIdentifier)
            PsiElement(identifier)('a')
            PsiElement(whitespace)(' ')
          PSASTWrapperElement(GenericIdentifier)
            PsiElement(identifier)('b')
            PsiElement(whitespace)(' ')
          PSASTWrapperElement(GenericIdentifier)
            PsiElement(identifier)('c')
            PsiElement(whitespace)(' ')
        PsiElement(=)('=')
        PsiElement(whitespace)(' ')
        PSASTWrapperElement(TypeConstructor)
          PSProperNameImpl(ProperName)
            PsiElement(proper name)('A')
            PsiElement(whitespace)('\n       ')
        PsiElement(|)('|')
        PsiElement(whitespace)(' ')
        PSASTWrapperElement(TypeConstructor)
          PSProperNameImpl(ProperName)
            PsiElement(proper name)('B')
            PsiElement(whitespace)(' ')
        PSASTWrapperElement(TypeAtom)
          PSASTWrapperElement(GenericIdentifier)
            PsiElement(identifier)('a')
            PsiElement(whitespace)('\n       ')
        PsiElement(|)('|')
        PsiElement(whitespace)(' ')
        PSASTWrapperElement(TypeConstructor)
          PSProperNameImpl(ProperName)
            PsiElement(proper name)('Email')
            PsiElement(whitespace)(' ')
        PSASTWrapperElement(TypeAtom)
          PSASTWrapperElement(TypeConstructor)
            PSASTWrapperElement(Qualified)
              PSProperNameImpl(ProperName)
                PsiElement(proper name)('HttpError')
                PsiElement(whitespace)(' ')
        PSASTWrapperElement(TypeAtom)
          PSASTWrapperElement(GenericIdentifier)
            PsiElement(identifier)('b')
            PsiElement(whitespace)('\n       ')
        PsiElement(|)('|')
        PsiElement(whitespace)(' ')
        PSASTWrapperElement(TypeConstructor)
          PSProperNameImpl(ProperName)
            PsiElement(proper name)('Boo')
            PsiElement(whitespace)(' ')
        PSASTWrapperElement(TypeAtom)
          PSASTWrapperElement(GenericIdentifier)
            PsiElement(identifier)('c')