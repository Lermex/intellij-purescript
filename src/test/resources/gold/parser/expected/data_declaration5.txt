Purescript File
  PSModuleDeclarationImpl(MODULE_DECLARATION)
    PsiElement(PSTokenType.module)('module')
    PSModuleNameImpl(MODULE_NAME)
      PsiElement(PSTokenType.PROPER_NAME)('A')
    PsiElement(PSTokenType.where)('where')
  PSDeclarationsImpl(DECLARATIONS)
    PSDeclarationImpl(DECLARATION)
      PSDataDeclarationImpl(DATA_DECLARATION)
        PsiElement(PSTokenType.data)('data')
        PSDeclarationTypeImpl(DECLARATION_TYPE)
          PsiElement(PSTokenType.PROPER_NAME)('User5')
        PSIdListImpl(ID_LIST)
          PSDeclarationIdImpl(DECLARATION_ID)
            PsiElement(PSTokenType.ID)('a')
          PSIdListImpl(ID_LIST)
            PSDeclarationIdImpl(DECLARATION_ID)
              PsiElement(PSTokenType.ID)('b')
            PSIdListImpl(ID_LIST)
              PSDeclarationIdImpl(DECLARATION_ID)
                PsiElement(PSTokenType.ID)('c')
        PsiElement(PSTokenType.=)('=')
        PSDataDeclarationItemListImpl(DATA_DECLARATION_ITEM_LIST)
          PSDataDeclarationItemImpl(DATA_DECLARATION_ITEM)
            PSTypeListImpl(TYPE_LIST)
              PSDeclarationTypeImpl(DECLARATION_TYPE)
                PsiElement(PSTokenType.PROPER_NAME)('A')
          PsiElement(PSTokenType.|)('|')
          PSDataDeclarationItemListImpl(DATA_DECLARATION_ITEM_LIST)
            PSDataDeclarationItemImpl(DATA_DECLARATION_ITEM)
              PSTypeListImpl(TYPE_LIST)
                PSDeclarationTypeImpl(DECLARATION_TYPE)
                  PsiElement(PSTokenType.PROPER_NAME)('B')
              PSIdListImpl(ID_LIST)
                PSDeclarationIdImpl(DECLARATION_ID)
                  PsiElement(PSTokenType.ID)('a')
            PsiElement(PSTokenType.|)('|')
            PSDataDeclarationItemListImpl(DATA_DECLARATION_ITEM_LIST)
              PSDataDeclarationItemImpl(DATA_DECLARATION_ITEM)
                PSTypeListImpl(TYPE_LIST)
                  PSDeclarationTypeImpl(DECLARATION_TYPE)
                    PsiElement(PSTokenType.PROPER_NAME)('Email')
                  PSTypeListImpl(TYPE_LIST)
                    PSDeclarationTypeImpl(DECLARATION_TYPE)
                      PsiElement(PSTokenType.PROPER_NAME)('HttpError')
                PSIdListImpl(ID_LIST)
                  PSDeclarationIdImpl(DECLARATION_ID)
                    PsiElement(PSTokenType.ID)('b')
              PsiElement(PSTokenType.|)('|')
              PSDataDeclarationItemListImpl(DATA_DECLARATION_ITEM_LIST)
                PSDataDeclarationItemImpl(DATA_DECLARATION_ITEM)
                  PSTypeListImpl(TYPE_LIST)
                    PSDeclarationTypeImpl(DECLARATION_TYPE)
                      PsiElement(PSTokenType.PROPER_NAME)('Boo')
                  PSIdListImpl(ID_LIST)
                    PSDeclarationIdImpl(DECLARATION_ID)
                      PsiElement(PSTokenType.ID)('c')